spring-cloud-kubernetes-client-autoconfig/src/main/java/org/springframework/cloud/kubernetes/client/KubernetesClientAutoConfiguration.java

```diff
@@ -42,12 +42,9 @@ public class KubernetesClientAutoConfiguration {
 
 	@Bean
 	@ConditionalOnMissingBean
-	public ApiClient apiClient(Environment environment) {
```

Comment:
the beauty of the previous clean-up that we did... We don't need this hack anymore, properties are correctly set. And we do already have tests that enforce this, and they don't fail once I made this change

Reply:
the method is still there, granted the argument has changed. And this is an auto-configuration class, we had removals like these in configurations classes before. I mean it's a method providing a bean, I kind of thought these are always safe to modify, since they are always public... If you disagree this would be the only change making this PR not possible to go to `main`

Reply:
of course, it would be easy to fix:

- `remove @Bean`
- add `@Deprecated(forRemoval=true)`
- add a new bean. 

:)

Reply:
Yes that is what I was going to suggest, just have the deprecated method call the new one.

We have been bitten by seemingness harmful changes, so I am trying to avoid that.

Reply:
I can cause myself a massive headache how someone can potentially abuse this method in their code, or I can deprecate it and have a PR that simply replaces it. I already maintain 4 PRs that can go only in the next major release, so my plan is this:

- leave this method as is, just add `@Deprecated` to it
- update one of the "next major release" PRs with the change where this one is dropped and a new one created.

It's simpler. Thx a lot for your feedback.